crates/codec/src/api/builder/template_builder.rs

```diff
@@ -1,110 +1,113 @@
-use svm_layout::Layout;
-use svm_types::Template;
+use svm_types::{
+    ApiSection, CodeSection, CtorsSection, DataSection, DeploySection, HeaderSection,
+    SchemaSection, SectionKind, SectionWrapper, Sections, Template,
+};
 
-use crate::template;
-
-/// Builds a raw representation for `deploy-template`
-/// Should be used for testing only.
-pub struct DeployTemplateBuilder {
-    version: Option<u16>,
-    name: Option<String>,
-    code: Option<Vec<u8>>,
-    layout: Option<Layout>,
-    ctors: Option<Vec<String>>,
+/// Builds a `Template`
+pub struct TemplateBuilder {
+    sections: Sections,
 }
 
-///
-/// # Example
-///  
-/// ```rust
-/// use std::io::Cursor;
-///
-/// use svm_types::Template;
-/// use svm_codec::api::builder::DeployTemplateBuilder;
-/// use svm_codec::template;
-///
-/// let layout = vec![5, 10].into();
-/// let ctors = vec!["init".to_string()];
-///
-/// let bytes = DeployTemplateBuilder::new()
-///            .with_version(0)
-///            .with_name("My Template")
-///            .with_code(&[0xC, 0x0, 0xD, 0xE])
-///            .with_layout(&layout)
-///            .with_ctors(&ctors)
-///            .build();
-///
-/// let mut cursor = Cursor::new(&bytes[..]);
-/// let actual = template::decode_deploy_template(&mut cursor).unwrap();
-///
-/// let expected = Template {
-///                  version: 0,
-///                  name: "My Template".to_string(),
-///                  code: vec![0xC, 0x0, 0xD, 0xE],
-///                  layout,
-///                  ctors: vec!["init".to_string()]
-///                };
-///
-/// assert_eq!(expected, actual);
-/// ```
-///
```

Comment:
I think we should keep this if possible. It'd still be relevant after updating it with the appropriate naming changes, right?